File: 2023-2024/sourcecode/mod3/lab9/RoomBasicsJava/app/src/main/java/com/example/roombasicsjava/ui/UserFragment.java
package com.example.roombasicsjava.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.roombasicsjava.data.dummydata.DataProvider;
import com.example.roombasicsjava.databinding.FragmentUserBinding;
import com.example.roombasicsjava.viewmodel.UserViewModel;

import java.util.List;

public class UserFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        com.example.roombasicsjava.databinding.FragmentUserBinding binding = FragmentUserBinding.inflate(inflater);

        UserViewModel userViewModel = new ViewModelProvider(this).get(UserViewModel.class);

        final UserAdapter userAdapter = new UserAdapter(new UserComparator());
        binding.rvList.setAdapter(userAdapter);
        binding.rvList.setLayoutManager(new LinearLayoutManager(requireActivity()));
        userViewModel.getUsers().observe(requireActivity(), userAdapter::submitList);

        binding.addButton.setOnClickListener(v -> {userViewModel.addUser(DataProvider.getUser());});
        binding.clearButton.setOnClickListener(v -> userViewModel.clearUsers());


        return binding.getRoot();
    }
}

File: 2023-2024/sourcecode/mod2/lab9/SharedPreferencesBasicsKotlin/app/src/main/java/com/example/sharedpreferencesbasicskotlin/ui/UserFragment.kt
package com.example.sharedpreferencesbasicskotlin.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import androidx.lifecycle.lifecycleScope
import com.example.sharedpreferencesbasicskotlin.data.dummydata.DataProvider
import com.example.sharedpreferencesbasicskotlin.databinding.FragmentUserBinding
import com.example.sharedpreferencesbasicskotlin.viewmodel.UserViewModel
import kotlinx.coroutines.flow.collectLatest
import kotlinx.coroutines.launch

class UserFragment : Fragment() {

     private lateinit var binding: FragmentUserBinding

    private val viewModel: UserViewModel by viewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentUserBinding.inflate(inflater)

        viewLifecycleOwner.lifecycleScope.launch {
            viewModel.username.collectLatest { user ->
                binding.textView.text = "Current saved user: $user"
            }
        }

        binding.addButton.setOnClickListener { viewModel.addUsername(DataProvider.username) }
        binding.clearButton.setOnClickListener { viewModel.clearUsername() }

        return binding.root
    }
}